src/compiler/types.ts

```diff
@@ -1525,7 +1526,9 @@ module ts {
     export interface SymbolLinks {
         target?: Symbol;                    // Resolved (non-alias) target of an alias
         type?: Type;                        // Type of value symbol
-        declaredType?: Type;                // Type of class, interface, enum, or type parameter
+        declaredType?: Type;                // Type of class, interface, enum, type alias, or type parameter
+        typeParameters?: TypeParameter[];   // Type parameters of type alias (undefined if non-generic)

```

Comment:
I think we either keep it this way or switch to a family of xxxSymbolLinks types like we have for types. But my preference would be to just keep it this way to reduce the noise.